### PR TITLE
adds companies and examples to motivation section

### PR DESCRIPTION
**Pull request by yati1998**
fixes issue #860

### CODE REVIEW

File: agile-development/intro-to-agile.md [hunk 1]
```diff
@@ -8,6 +8,21 @@
...
+- Stakeholder Engagement
+- Transparency
+- Early and Predictable Delivery
+- Predictable Costs and Schedule
+- Allows for Change
+- Focuses on Business Value
+[(seguetech.com)](https://www.seguetech.com/8-benefits-of-agile-software-development/)
+
+**Which companies use agile development?**
+- IBM
```
Comment (alodahl) [inline agile-development/intro-to-agile.md]: provide links to their sites, please

File: data-structures/intro-to-data-structures.md [hunk 2]
```diff
@@ -38,6 +38,8 @@ Linear data structures arrange values in a linear form, and some commonly used l
 **Graphs**: a general branching structure with less strict conditions than a tree.
 Some other data structures include record, union, tagged union, class, graphs, and binary trees.
 
+Data Structure forms the major part of IT companies. Many top companies like Microsoft and google hire on the basis of coding skills that mainly includes data structure.
```
Comment (alodahl) [inline data-structures/intro-to-data-structures.md]: How do  Microsoft and *G*oogle use data structures, though?

File: functional-programming/FP.md [hunk 3]
```diff
@@ -21,6 +21,10 @@
 ### Motivation
  __Functional Programming__ (FP) is a programming paradigm that's different from object-oriented programming. Functional programming is all about separating data from behavior, so that software is __more predictable, more robust, and easier to test__. Certain tools become readily available with functional programming, because __functions can be used as data__, which means they can be passed around as parameters, and used as inputs into other functions. This can make code more concise and easier to read.
 
+ **Which companies use Functional Programming?**
+ - Skyscanner(JVM)
```
Comment (alodahl) [inline functional-programming/FP.md]: needs links and examples

File: jquery/jquery.md [hunk 4]
```diff
@@ -15,6 +15,12 @@ Approximately 2.5 hours
 ### Motivation
 jQuery is a popular and easy-to-learn JavaScript library that allows you to manipulate HTML elements. jQuery allows developers to quickly make webpages interactive across different browsers. 
 
+**Which companies use jQuery?**
+Below is the list companies using jQuery:
+- Uber
```
Comment (alodahl) [inline jquery/jquery.md]: links and examples

File: testing-and-tdd/jasmine-testing.md [hunk 5]
```diff
@@ -14,6 +14,13 @@ Here are links to lessons that should be completed before this lesson:
 
 Learn to use more than one testing tool for flexibility.
 
+Jasmine is a Behavior Driven Development testing framework for JavaScript. It does not rely on browsers, DOM, or any JavaScript framework. Thus it's suited for websites, Node.js projects, or anywhere that JavaScript can run.[(stackshare.io)](https://stackshare.io/jasmine)
+
+**Which companies use Jasmine testing?**
+- Coderus
```
Comment (alodahl) [inline testing-and-tdd/jasmine-testing.md]: links please 
no examples needed

File: testing-and-tdd/testing-and-tdd.md [hunk 6]
```diff
@@ -15,6 +15,11 @@ About 2 hours
 ### Motivation
 Testing makes your code better, lets you work faster, and can actually be fun!
 
+**Which companies use automated testing?**
+- Testbytes: The enviable track record of [providing test solutions](https://www.testbytes.net/software-testing-services/) and services on time has helped them to be one of the leading testing companies in India.
```
Comment (alodahl) [inline testing-and-tdd/testing-and-tdd.md]: 👏🏼

File: testing-and-tdd/testing-and-tdd.md [hunk 7]
```diff
@@ -15,6 +15,11 @@ About 2 hours
...
+- Accenture: The testing team at Accenture assists the client companies to launch some new technology in this fast-paced world. This is owing to the testing team ability to help companies be sure of the product quality delivered by them as well as offer a seamless customer experience.
```
Comment (alodahl) [inline testing-and-tdd/testing-and-tdd.md]: needs accenture site link

**Comment by yati1998** [comment]
@alodahl I have made the required changes. Please do review them

File: functional-programming/FP.md [hunk 8]
```diff
@@ -21,6 +21,12 @@
 ### Motivation
  __Functional Programming__ (FP) is a programming paradigm that's different from object-oriented programming. Functional programming is all about separating data from behavior, so that software is __more predictable, more robust, and easier to test__. Certain tools become readily available with functional programming, because __functions can be used as data__, which means they can be passed around as parameters, and used as inputs into other functions. This can make code more concise and easier to read.
 
+ **Which companies use Functional Programming?**
+ - Twitter uses Scala, an FP with Java syntax running on the JVM. [How and why Twitter uses Scala.](https://www.redfin.com/blog/how_and_why_twitter_uses_scala)
```
[the remark this answers is not in the document]
Comment (yati1998) [reply functional-programming/FP.md]: As mentioned in the description Scala is an FP having java syntax.
Clojure is also a FP.
these are mentioned in the site from where got these examples. and also sited at the end.

**Comment by yati1998** [comment]
@alodahl I have made the changes. Please do review them

File: testing-and-tdd/jasmine-testing.md [hunk 9]
```diff
@@ -14,6 +14,13 @@ Here are links to lessons that should be completed before this lesson:
 
 Learn to use more than one testing tool for flexibility.
 
+Jasmine is a Behavior Driven Development testing framework for JavaScript. It does not rely on browsers, DOM, or any JavaScript framework. Thus it's suited for websites, Node.js projects, or anywhere that JavaScript can run.[(stackshare.io)](https://stackshare.io/jasmine)
```
Comment (saberprashant) [inline testing-and-tdd/jasmine-testing.md]: Awesome 😄

File: functional-programming/FP.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (alodahl) [reply functional-programming/FP.md]: Please provide info here, this is not the place to expect learners to search around a link unless they want to. HOW do the use functional programming in a way that helps their business?

**Comment by yati1998** [comment]
@alodahl I have made the changes. Please do review them.

File: data-structures/intro-to-data-structures.md (hunk 2 above)
Comment (alodahl) [reply data-structures/intro-to-data-structures.md]: *G*oogle

File: data-structures/intro-to-data-structures.md [hunk 10]
```diff
@@ -38,6 +38,11 @@ Linear data structures arrange values in a linear form, and some commonly used l
...
+
+Example of people getting hired having data structure knowledge:
+- [How I landed offers from microsoft, amozon and twitter (by Zhia Hwa Chong)](https://www.freecodecamp.org/news/how-i-landed-offers-from-microsoft-amazon-and-twitter-without-an-ivy-league-degree-d62cfe286eb8/)
```
Comment (alodahl) [inline data-structures/intro-to-data-structures.md]: Microsoft, Amazon and Twitter

File: jquery/jquery.md [hunk 11]
```diff
@@ -15,6 +15,13 @@ Approximately 2.5 hours
 ### Motivation
 jQuery is a popular and easy-to-learn JavaScript library that allows you to manipulate HTML elements. jQuery allows developers to quickly make webpages interactive across different browsers. 
 
+**Which companies use jQuery?**
+Below is the list companies using jQuery:
+- [github](http://github.com/)
```
Comment (alodahl) [inline jquery/jquery.md]: GitHub
Stack Overflow

File: functional-programming/FP.md [hunk 12]
```diff
@@ -21,6 +21,14 @@
...
 
+ **Which companies use Functional Programming?**
+ - [Bloomberg uses Ocmal which is a functional programming language](http://cufp.org/2014/maxime-ransan-adopting-functional-programming-with-ocaml-at-bloomberg-lp.html)
+   - Bloomberg utilizes functional programming (OCaml) in production for powering part of its Bloomberg Professional Services product. Aided by a collaboration with LexiFi, several Bloomberg applications are now making use of GUI generation and financial contract representation; Bloomberg furthermore has embraced OCaml to develop domain specific languages and the problem of integrating the vast amount of data available in the Bloomberg infrastructure.[(cufp.org)](http://cufp.org/2014/maxime-ransan-adopting-functional-programming-with-ocaml-at-bloomberg-lp.html)
```
Comment (alodahl) [inline functional-programming/FP.md]: nice, thank you

**Comment by yati1998** [comment]
@alodahl I have modified the companies name. Please do review them. Thank you.

**Comment by alodahl** [comment]
@saberprashant are you ok with the changes?

**Pull request merged**